wp-tevko-responsive-images.php

```diff
@@ -265,31 +265,76 @@ function tevkori_get_src_sizes( $id, $size = 'thumbnail' ) {
 }
 
 /**
- * Filter for extending image tag to include srcset attribute.
+ * Filter for the_content to add sizes and srcset attributes to images.
  *
- * @see images_send_to_editor
- * @return string HTML for image.
+ * @since 3.0
+ *
+ * @param string $content The raw post content to be filtered.
  */
-function tevkori_extend_image_tag( $html, $id, $caption, $title, $align, $url, $size, $alt ) {
-	add_filter( 'editor_max_image_size', 'tevkori_editor_image_size' );
-
-	$sizes = tevkori_get_sizes( $id, $size );
+function tevkori_filter_content_images( $content ) {
+	return preg_replace_callback(
+		'/<img\s([^>]+)>/i',
+		
+		// Don't use an anonymous callback function because it isn't supported by PHP 5.2.
+		tevkori_filter_content_images_callback( $matches ),
+		$content
+	);
+}
+add_filter( 'the_content', 'tevkori_filter_content_images', 999, 1 );
 
-	// Build the data-sizes attribute if sizes were returned.
-	if ( $sizes ) {
-		$sizes = 'data-sizes="' . $sizes . '"';
-	}
+/**
+ * Callback function for tevkori_filter_content_images.
+ *
+ * @since 3.0
+ *
+ * @see tevkori_filter_content_images
+ * @param array $matches Array containing the regular expression matches.
+ */
+function tevkori_filter_content_images_callback( $matches ) {
+	$atts = $matches[1];
+	$sizes = $srcset = '';
+
+	// Get the value of the class attribute.
+	preg_match( '/class="([^"]+)"/i', $atts, $classes );
+
+	if ( $classes ) {
+		
+		// Grab ID and size info from core classes.
+		preg_match( '/wp-image-([0-9]+)/i', $classes[1], $id );
+		preg_match( '/size-([^\s|"]+)\s|"/i', $classes[1], $size );
+
+		$id = (int) $id[1];
+
+		// If a class with size name is present, use it.
+		if ( $size ) {
+			$size = $size[1];
+			
+		// Otherwise create an array with the values from the width and height attributes.
```

Comment:
@jaspermdegroot there are often cases where the developer chooses to remove the WordPress classes and w/h attributes. Would we be able to retrieve the image ID if those weren't present? Or does this function fire with a higher priority, so that our calculations happen before theme customizations kick in?